### DISCUSSION TITLE
NaN in synthetic likelihood

### DISCUSSION THREAD

**Discussion by Fuhan-Yang** [kingaa/pomp#207] [Q&A] [answered]
Hi Dr.King, 

I have a quick question about synthetic likelihood. When I summarized the information of probes using `summary(probe(pomp, probe_list,MLE))`, some of the models provided NaN in the synthetic likelihood. Do you happen to know what can be the causes? Thank you!

**Comment by kingaa** [comment]
The synthetic likelihood is inversely proportional to the variance of the simulated probes. If the variance matrix has a zero eigenvalue, this will lead to an `NaN`.  Can you check this?

**Comment by Fuhan-Yang** [comment]
There is no zero eigenvalue in the variance matrix of the simulated probes. But the eigenvalues are very dispersed, ranging from -2e-5 to 8e9. plot of the expit transformed of the eigenvalues looks like this:
![Screenshot 2024-03-19 at 8 44 45 AM](https://github.com/kingaa/pomp/assets/59420891/2c8758c0-dc35-4888-9e22-ba27f80cc710)

The magnitude of the simulated probes ranges from 1e-3 to 1e4, and the one from the data ranges from 1e-2 to 1e3. Could this be the reason?

**Comment by kingaa** [comment, marked as answer]
Yes, I think the variance-covariance matrix is ill-conditioned.  You may be able to resolve this issue by re-scaling the probes so that their variability is more of unit magnitude.  Also, if some probes are redundant, i.e., they are almost perfectly well predicted by some combination of other probes, this can lead to ill-conditioning.

**Reply by Fuhan-Yang** [reply]
Thanks for the reply. After checking the probes, it is one of the summary statistics that always returned the same value for all the simulated probes may cause this issue (other models that didn't return the same value don't have NA synthetic likelihood). Could this also be the reason?

**Reply by kingaa** [reply]
Yes, if there is zero variance in one of the probes, then the inverse will have a zero eigenvalue in that direction.

**Reply by Fuhan-Yang** [reply]
Got it. Thanks a lot!
